tests/unit_tests/run_models/test_base_run_model.py

```diff
@@ -109,14 +113,61 @@ def test_check_if_runpath_exists(
     )
     simulation_arguments.num_iterations = number_of_iterations
 
-    def get_run_path_mock(realizations, iteration=None):
-        if iteration is not None:
-            return [f"out/realization-{r}/iter-{iteration}" for r in realizations]
-        return [f"out/realization-{r}" for r in realizations]
+    model_config = ModelConfig(runpath_format_string=run_path)
+    subs_list = SubstitutionList()
+    config = MagicMock()
+    config.model_config = model_config
+    config.substitution_list = subs_list
 
-    brm = BaseRunModel(simulation_arguments, MagicMock(), None, None, None, None)
-    brm.run_paths.get_paths = get_run_path_mock
+    brm = BaseRunModel(simulation_arguments, config, None, None, None, None)
     brm.facade = MagicMock(
         run_path=run_path,
     )
     assert brm.check_if_runpath_exists() == expected
+
+
+@pytest.mark.usefixtures("use_tmpdir")
+@pytest.mark.parametrize(
+    "run_path_format", ["realization-<IENS>/iter-<ITER>", "realization-<IENS>"]
+)
+@pytest.mark.parametrize(
+    "active_realizations", [[True], [True, True], [True, False], [False], [False, True]]
+)
+def test_delete_run_path(run_path_format, active_realizations):
+    simulation_arguments = EnsembleExperimentRunArguments(
+        random_seed=None,
+        active_realizations=active_realizations,
+        current_case=None,
+        target_case=None,
+        start_iteration=0,
+        iter_num=0,
+        minimum_required_realizations=0,
+        ensemble_size=1,
+        stop_long_running=False,
+    )
+    expected_remaining = []
+    expected_removed = []
+    for iens, mask in enumerate(active_realizations):
```

Comment:
Not sure I like this logic, could fix it by putting everything in one parameterize, but then there would be a lot of repetition there, what do you think?

Reply:
fine with me.